### PR TITLE
feat: provide driver explicitly

### PR DESCRIPTION
**Pull request by tsaleksandrova**

### CODE REVIEW

File: spec/directConnectionProvider.spec.js [hunk 1]
```diff
@@ -1,76 +1,103 @@
...
+            latestVersionUrl: mockUrl + '/LATEST_RELEASE_{chrome.latest}',
+            url: mockUrl + '/{chrome.latest}/chromedriver_win32.zip',
+          },
+        chromedriverLocal:
+          {
+            localPath: path.join(__dirname, "directConnectionProvider/mockChromeDriver.js"),
+            version: '{latest}',
+            filename: 'chromedriver',
+            executable: 'chromedriver-{latest}',
+            baseurl: mockUrl,
```
Comment (maximnaidenov) [inline spec/directConnectionProvider.spec.js]: I suggest to leave only the relevant params for this usecase, this is a form of documentation and a test that no other params are used unintentionally.

File: spec/directConnectionProvider/mockChromeDriver.js [hunk 2]
```diff
@@ -0,0 +1 @@
+console.log("mock");
```
Comment (maximnaidenov) [inline spec/directConnectionProvider/mockChromeDriver.js]: if this mocks the executable and is actually called, it needs the hashbang on the first line

File: src/connection/directConnectionProvider.js [hunk 3]
```diff
@@ -160,7 +166,7 @@ DirectConnectionProvider.prototype.setupEnv = function() {
   if (!that.seleniumConfig.address) {
     // start local driver
     if (that.seleniumConfig.useSeleniumJarFlag) {
-      promises.push(this._downloadBinary(that.binaries['selenium']).then(
+      promises.push(this._getBinaryFileName(BINARIES.SELENIUM).then(
```
Comment (maximnaidenov) [inline src/connection/directConnectionProvider.js]: the name fo the method is misleading as the parameter is not the filename (filename is provided in 'filename' property) but the name of the configuration !

File: spec/directConnectionProvider.spec.js (hunk 1 above)
Comment (tsaleksandrova) [reply spec/directConnectionProvider.spec.js]: and also a test that no instance params are modified in a wrong way

File: src/connection/directConnectionProvider.js (hunk 3 above)
Comment (tsaleksandrova) [reply src/connection/directConnectionProvider.js]: for a given configuration name, it gives the path to the executable
(and if it's not found, an archive/executable is downloaded, renamed if needed, and then the new path is returned) 
i agree there is a clash with the 'filename' param, which is the desired name of a downloaded executable (after renaming)

**Pull request merged**